nano/secure/blockstore.hpp

```diff
@@ -16,6 +16,62 @@
 
 namespace nano
 {
+class block_details
+{
+	static_assert (std::is_same<std::underlying_type<nano::epoch>::type, uint8_t> (), "Epoch enum is not the proper type");
+	static_assert (static_cast<uint8_t> (nano::epoch::max) < (1 << 5), "Epoch max is too large for the sideband");
+
+public:
+	block_details () = default;
+	block_details (nano::epoch const epoch_a, bool const is_send_a, bool const is_receive_a, bool const is_epoch_a) :
+	epoch (epoch_a), is_send (is_send_a), is_receive (is_receive_a), is_epoch (is_epoch_a)
+	{
+	}
+	static constexpr size_t size ()
+	{
+		return 1;
+	}
+	bool operator== (block_details const & other_a) const
+	{
+		return epoch == other_a.epoch && is_send == other_a.is_send && is_receive == other_a.is_receive && is_epoch == other_a.is_epoch;
+	}
+	void serialize (nano::stream &) const;
+	bool deserialize (nano::stream &);
```

Comment:
Some functions are defined inline in the class definition but `serialize`/`deserialize` aren't. Would be good to move them all out, if only for consistency.